cpp/src/sampling/random_walks.cuh

```diff
@@ -690,11 +662,31 @@ struct random_walker_t {
 
   device_vec_t<edge_t> const& get_out_degs(void) const { return d_cached_out_degs_; }
 
+  vertex_t get_vertex_padding_value(void) const { return vertex_padding_value_; }
+
+  weight_t get_weight_padding_value(void) const { return weight_padding_value_; }
+
+  void init_padding(device_vec_t<vertex_t>& d_coalesced_v,
+                    device_vec_t<weight_t>& d_coalesced_w) const
+  {
+    thrust::fill(rmm::exec_policy(handle_.get_stream())->on(handle_.get_stream()),
+                 d_coalesced_v.begin(),
+                 d_coalesced_v.end(),
+                 vertex_padding_value_);
+
+    thrust::fill(rmm::exec_policy(handle_.get_stream())->on(handle_.get_stream()),
+                 d_coalesced_w.begin(),
+                 d_coalesced_w.end(),
+                 weight_padding_value_);
+  }
+
  private:
   raft::handle_t const& handle_;
   index_t num_paths_;
   index_t max_depth_;
   device_vec_t<edge_t> d_cached_out_degs_;
+  vertex_t const vertex_padding_value_;
+  weight_t const weight_padding_value_;
```

Comment:
Better add trailing {} to class member variables (unless the default constructor is deleted).

Reply:
Why? Those are `const` members. No chance their initialization could be missed in the cnstr. initalization list. That would trigger a compiler error.

In general, it might be helpful to differentiate between _personal_ code-style (which may differ from reviewer to reviewer, hence they should not / cannot be enforced) and generic code-style, or actual issues, missuses, mistakes (which is obvious that they need correcting).

Reply:
In fact, with brace default initializer, '{}` on the `const` members things are worse. Because, in that case, the constructor initialization list is allowed to _miss_ the initialization of `const` members. Which is _not_ an option here. The constructor must decide what value to use for those `const`'s, because the logic is not simply: "default to 0".

Reply:
Oh, I didn't see const here. I was commenting about the private member variables in general.